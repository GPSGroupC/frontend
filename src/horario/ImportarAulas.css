/**
 * CSS PARA EL COMPONENTE IMPORTAR AULAS
 */