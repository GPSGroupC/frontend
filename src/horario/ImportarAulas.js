import React, { useState, Fragment } from "react";
import { nanoid } from "nanoid";
import data from "../mock-data.json";
import FilaReadOnly from "./componentes/FilaReadOnlyAulas";
import FilaEditable from "./componentes/FilaEditableAulas";
import { Link } from 'react-router-dom'
import eina from '../images/eina-logo.png'
import './PantallaHorarios.css';
import Button from 'react-bootstrap/Button';
import ButtonGroup from 'react-bootstrap/Button';
import Api from "./servicios/api";

function ImportarAulas() {

  async function importarAulas(formdata) {
    await Api.importarAulas(formdata).then(r => {
      console.log(r.status);
      if (r.status === 200) {
        document.getElementById("file").value = "";
        window.alert("Aulas importadas con éxito");
      } else {
        window.alert("Ha ocurrido un error al importar las aulas");
      }
    }).catch(err => {
      console.log("Error al importar aulas: ", err)
    })
  }

  const handleSubmit = (e) => {
    e.preventDefault();
    console.log(e.target[0].files[0]);
    const answer = window.confirm("¿Estás seguro? Se perderán todas las aulas importadas previamente");
    if (answer) {
      const formdata = new FormData()
      formdata.append('file', e.target[0].files[0]);
      formdata.append('conservarNoImportadas', document.getElementById("conservarAulas").checked);
      importarAulas(formdata);
    } else {


    }
  };


  return (

    <div>
      <div id="import-box">
        <h3>
          Importar Aulas
        </h3><br></br><br></br>
        <p>
          Selecciona el archivo excel para importar las aulas
        </p><br></br><br></br>
        <div id="fileform">
          <form onSubmit={handleSubmit}>
            <label>
              <input id="file" type="file" name="file" />
            </label>
            <input id="importarbutton" type="submit" value="Importar" />
          </form>
          <br></br><br></br>
          <label>
            <input id="conservarAulas" type="checkbox" />
            Conservar aulas que se habían añadido manualmente
          </label>
        </div>
      </div>
    </div>
  );
};

export default ImportarAulas;
