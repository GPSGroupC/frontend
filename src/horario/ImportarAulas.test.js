
import React, { useState, Fragment } from "react";
import ReactDOM from "react-dom";
import ImportarAulas from "./ImportarAulas";
import renderer from 'react-test-renderer';



describe("Snapshot testing", () => {

test('Se renderiza correctamente', () => {

    const rendererComponent = renderer.create(<ImportarAulas></ImportarAulas>).toJSON()
    expect(rendererComponent).toMatchSnapshot()

})
})
