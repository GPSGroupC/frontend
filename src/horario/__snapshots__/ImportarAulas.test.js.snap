// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing Se renderiza correctamente 1`] = `
<div>
  <div
    id="import-box"
  >
    <h3>
      Importar Aulas
    </h3>
    <br />
    <br />
    <p>
      Selecciona el archivo excel para importar las aulas
    </p>
    <br />
    <br />
    <div
      id="fileform"
    >
      <form
        onSubmit={[Function]}
      >
        <label>
          <input
            id="file"
            name="file"
            type="file"
          />
        </label>
        <input
          id="importarbutton"
          type="submit"
          value="Importar"
        />
      </form>
      <br />
      <br />
      <label>
        <input
          id="conservarAulas"
          type="checkbox"
        />
        Conservar aulas que se habían añadido manualmente
      </label>
    </div>
  </div>
</div>
`;
